gateway-service/src/main/java/org/zowe/apiml/gateway/services/ServicesInfoService.java

```diff
@@ -0,0 +1,269 @@
+
+        if (application == null) {
+            return ServiceInfo.builder()
+                    .serviceId(serviceId)
+                    .status(InstanceInfo.InstanceStatus.UNKNOWN)
+                    .build();
+        }
+
+        return getServiceInfo(application);
+    }
+
+    private ServiceInfo getServiceInfo(Application application) {
+        String serviceId = application.getName().toLowerCase();
+
+        List<InstanceInfo> appInstances = application.getInstances();
+        if (appInstances == null || appInstances.isEmpty()) {
+            return ServiceInfo.builder()
+                    .serviceId(serviceId)
+                    .status(InstanceInfo.InstanceStatus.DOWN)
+                    .build();
+        }
+
+        return ServiceInfo.builder()
+                .serviceId(serviceId)
+                .status(getStatus(appInstances))
+                .apiml(getApiml(appInstances))
+                .instances(getInstances(appInstances))
+                .build();
+    }
+
+    private ServiceInfo.Apiml getApiml(List<InstanceInfo> appInstances) {
+        return ServiceInfo.Apiml.builder()
+                .apiInfo(getApiInfos(appInstances))
+                .service(getService(appInstances))
+                .authentication(getAuthentication(appInstances))
+                .build();
+    }
+
+    private List<ServiceInfo.ApiInfoExtended> getApiInfos(List<InstanceInfo> appInstances) {
+        List<ServiceInfo.ApiInfoExtended> completeList = new ArrayList<>();
+
+        for (InstanceInfo instanceInfo : appInstances) {
+            List<ApiInfo> apiInfoList = eurekaMetadataParser.parseApiInfo(instanceInfo.getMetadata());
+            completeList.addAll(apiInfoList.stream()
+                    .map(apiInfo -> ServiceInfo.ApiInfoExtended.builder()
+                            .apiId(apiInfo.getApiId())
+                            .basePath(getBasePath(apiInfo, instanceInfo))
+                            .baseUrl(getBaseUrl(apiInfo, instanceInfo))
+                            .gatewayUrl(apiInfo.getGatewayUrl())
+                            .swaggerUrl(getGatewayUrl(
+                                    apiInfo.getSwaggerUrl(),
+                                    instanceInfo.getAppName().toLowerCase(),
+                                    ServiceType.API,
+                                    eurekaMetadataParser.parseRoutes(instanceInfo.getMetadata())
+                            ))
+                            .documentationUrl(apiInfo.getDocumentationUrl())
+                            .version(apiInfo.getVersion())
+                            .isDefaultApi(apiInfo.isDefaultApi())
+                            .build())
+                    .collect(Collectors.toList()));
+        }
+
+        return filterByIdAndMajorVersion(completeList);
+    }
+
+    private ServiceInfo.Service getService(List<InstanceInfo> appInstances) {
+        InstanceInfo instanceInfo = appInstances.get(0);
```

Comment:
What about making a separate method for getting the highest version and then convert?